specification/migrate/resource-manager/Microsoft.Migrate/2017-11-11-preview/migrate.json

```diff
@@ -0,0 +1,2940 @@
+          "readOnly": true
+        },
+        "datacenterManagementServer": {
+          "type": "string",
+          "description": "Name of the server hosting the datacenter management solution.",
+          "readOnly": true
+        },
+        "datacenterMachineId": {
+          "type": "string",
+          "description": "ID of the machine as tracked by the datacenter management solution.",
+          "readOnly": true
+        },
+        "datacenterManagementServerId": {
+          "type": "string",
+          "description": "ID of the server hosting the datacenter management solution.",
+          "readOnly": true
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the machine",
+          "readOnly": true
+        },
+        "displayName": {
+          "type": "string",
+          "description": "User readable name of the machine as defined by the user in their private datacenter.",
+          "readOnly": true
+        },
+        "megabytesOfMemory": {
+          "type": "number",
+          "description": "Memory in Megabytes.",
+          "readOnly": true
+        },
+        "numberOfCores": {
+          "type": "integer",
+          "description": "Processor count.",
+          "readOnly": true
+        },
+        "operatingSystem": {
+          "type": "string",
+          "description": "Operating System of the machine.",
+          "readOnly": true
+        },
+        "groups": {
+          "type": "array",
+          "description": "List of references to the groups that the machine is member of.",
+          "readOnly": true,
+          "items": {
+            "type": "string",
+            "readOnly": true,
+            "description": "Reference to a group that this machine is member of. /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/projects/{projectName}/groups/{groupName}"
+          }
+        },
+        "createdTimestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Time when this machine was created. Date-Time represented in ISO-8601 format.",
+          "readOnly": true
+        },
+        "updatedTimestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Time when this machine was last updated. Date-Time represented in ISO-8601 format.",
+          "readOnly": true
+        },
+        "discoveredTimestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Time when this machine was discovered by Azure Migrate agent. Date-Time represented in ISO-8601 format.",
+          "readOnly": true
+        },
+        "disks": {
+          "type": "object",
+          "description": "Dictionary of disks attached to the machine. Key is ID of disk. Value is a disk object",
+          "readOnly": true,
+          "properties": {
```

Comment:
>properties [](start = 11, length = 10)

It is bet not to define objects inline, so you can get a better code generation

Reply:
Its a dictionary, so we should not create a separate object for this.
Also, https://swagger.io/docs/specification/data-models/dictionaries/ is applicable for OpenAPI 3 only, while Azure-Rest-api-specs is version 2.0.
I think its ok to define dictionary inline.

Reply:
It can be defined like in the link. For example, here is an example from another RP:
https://github.com/Azure/azure-rest-api-specs/blob/0bef2b99ad6d52781e3c308bc6b420f6eef59ef7/specification/network/resource-manager/Microsoft.Network/stable/2017-09-01/network.json#L117-L123